### PR TITLE
Rolling restarts done via cronjob

### PR DESCRIPTION
**Pull request by rustyShacklefurd**
Signed-off-by: Matt Halder <rustyShacklefurd@users.noreply.github.com>

**Description**:
This PR adds the ability to enable or disable a rolling restart of the hedera-json-rpc-relay deployment

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: helm-chart/templates/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.rolling_restart.enabled }}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "json-rpc-relay.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  concurrencyPolicy: Forbid
+  schedule: '0 0 * * *'
```
Comment (Nana-EC) [inline helm-chart/templates/cronjob.yaml:9]: Can we go with `@daily` or `@midnight` to make this more readable

File: helm-chart/values.yaml [hunk 2]
```diff
@@ -95,9 +95,14 @@ config:
...
   # hosted:
   #   HEDERA_NETWORK: ""
-secret:
   OPERATOR_ID_MAIN: ""
   OPERATOR_KEY_MAIN: ""
   OPERATOR_ID_ETH_SENDRAWTRANSACTION: ""
-  OPERATOR_KEY_ETH_SENDRAWTRANSACTION: ""  
+  OPERATOR_KEY_ETH_SENDRAWTRANSACTION: ""
+
+# Enable rolling_restarts if SDK calls fail this is usually due to stale connections that get cycled on restart
```
Comment (Nana-EC) [inline helm-chart/values.yaml:103]: Can we change the replicaCount to 2 with this PR?
Or would k8s handle this by spinning one pod up first and moving traffic there before dropping the current pod?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
LG, some clarifying questions

File: helm-chart/templates/cronjob.yaml (hunk 1 above)
Comment (rustyShacklefurd) [reply helm-chart/templates/cronjob.yaml]: For this I have variablized the `rolling_restart.schedule` and the default is set to `@daily` in values.yaml

File: helm-chart/environments/minikube.yaml [hunk 3]
```diff
@@ -8,10 +8,12 @@ config:
...
-
-secret:
   OPERATOR_ID_MAIN: ""
   OPERATOR_KEY_MAIN: ""
   OPERATOR_ID_ETH_SENDRAWTRANSACTION: ""
   OPERATOR_KEY_ETH_SENDRAWTRANSACTION: ""
 
+rolling_restart:
+  enabled: true
+  schedule: "0 0 * * *"
```
Comment (Nana-EC) [inline helm-chart/environments/minikube.yaml:18]: If it lets you
```suggestion
  schedule: "@daily"
```

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**